src/components/Button.tsx

```diff
@@ -0,0 +1,17 @@
+import "./button.scss"
+import {MouseEventHandler} from "react";
+
+type ButtonProps = {
+  buttonType: 'button' | 'submit' | 'reset' | undefined;
+  buttonText: string;
+  onClick: MouseEventHandler <HTMLButtonElement> | undefined;
```

Comment:
just mark onClick as not required property, no need to pass here undefined in explicit form:

onClick?: ...